src/containerapp/azext_containerapp/custom.py

```diff
@@ -2197,3 +2201,108 @@ def show_env_managed_identity(cmd, name, resource_group_name):
         r["identity"] = {}
         r["identity"]["type"] = "None"
         return r["identity"]
+
+
+def list_java_components(cmd, environment_name, resource_group_name):
+    raw_parameters = locals()
+    java_component_decorator = JavaComponentDecorator(
+        cmd=cmd,
+        client=JavaComponentPreviewClient,
+        raw_parameters=raw_parameters,
+        models=CONTAINER_APPS_SDK_MODELS
+    )
+    return java_component_decorator.list()
+
+
+def show_java_component(cmd, java_component_name, environment_name, resource_group_name, target_java_component_type):
+    raw_parameters = locals()
+    java_component_decorator = JavaComponentDecorator(
+        cmd=cmd,
+        client=JavaComponentPreviewClient,
+        raw_parameters=raw_parameters,
+        models=CONTAINER_APPS_SDK_MODELS
+    )
+    result = java_component_decorator.show()
+
+    current_type = safe_get(result, "properties", "componentType")
+    if current_type and target_java_component_type.lower() != current_type.lower():
+        raise ResourceNotFoundError(f"(JavaComponentNotFound) JavaComponent '{java_component_name}' was not found.")
+
+    return result
+
+
+def delete_java_component(cmd, java_component_name, environment_name, resource_group_name, target_java_component_type, no_wait):
+    raw_parameters = locals()
+    java_component_decorator = JavaComponentDecorator(
+        cmd=cmd,
+        client=JavaComponentPreviewClient,
+        raw_parameters=raw_parameters,
+        models=CONTAINER_APPS_SDK_MODELS
+    )
+
+    try:
+        result = java_component_decorator.show()
+    except CLIInternalError as e:
+        return
```

Comment:
why returning all internal errors? why do you have to show first?

Reply:
Yes,  firstly it cannot `return` here.
And is the show required?
What if call `java_component_decorator.delete()` directly? Or it is for more friendly error message for customer?

Reply:
I test if remove the `show` logic before `delete`, I remove show is better.

Because currently, it cannot show customer if the resource group or the environment not exist.

Here is the behavior after removing the `show` logic before `delete`:
```
 azure-cli-extensions\src\containerapp> az containerapp env java-component spring-cloud-eureka delete -g a --environment b -n c
(ResourceGroupNotFound) Resource group 'a' could not be found.

azure-cli-extensions\src\containerapp> az containerapp env java-component spring-cloud-eureka delete -g arc-appint-forxinyu12-rg0 --environment b -n c
(ResourceNotFound) The Resource 'Microsoft.App/managedEnvironments/b' under resource group 'arc-appint-forxinyu12-rg0' was not found. For more details please go to https://aka.ms/ARMResourceNotFoundFix

azure-cli-extensions\src\containerapp> az containerapp env java-component spring-cloud-eureka delete -g arc-appint-forxinyu12-rg0 --environment envforregistry -n c
Command group 'containerapp env java-component' is in preview and under development. Reference and support levels: https://aka.ms/CLI_refstatus
Are you sure you want to perform this operation? (y/n): y
azure-cli-extensions\src\containerapp> 
```